### DISCUSSION TITLE
File Uploader: Estados de "cargando" y de archivo cargado.

### DISCUSSION THREAD

**Discussion by AleMCortez** [TiendaNube/nimbus-design-system#199] [Q&A] [answered]
Buenas!

En el desarrollo del New Checkout tenemos [este figma](https://www.figma.com/file/0EOCneK1m7q9IQPBDxeBJe/%F0%9F%A7%BE-New-Billing-Engine?node-id=1336%3A237383&mode=dev) que especifica este diseño:

<img width="782" alt="image" src="https://github.com/TiendaNube/nimbus-design-system/assets/90856893/b55ac8ff-d984-475f-83f9-a73ff2d1ce9f">

Lo importante es el componente que dice "Elegir comprobante" y nos deja subir un archivo. Para esto estuve usando el componente `FileUploader` de nimbus, y pude replicar lo que seria el primer paso del diseño sin problema. El problema es con los pasos siguientes, que piden que el componente se muestre en estado de carga mientras se cargae el archivo y luego que muestre el nombre del archivo y un boton para borrarlo cuando termina de cargarse. No encuentro forma de lograr esto en la [documentación del componente](https://nimbus.tiendanube.com/es-AR/documentation/atomic-components/file-uploader).

Nimbus ahora mismo no proporciona una manera simple de lograr esto? Si no, saben si hay planes de desarrollarlo proximamente?

**Comment by juanchigallego** [comment, marked as answer]
Buenas!

Sólo para dejar registro acá de lo que hablamos por Slack, vamos a avanzar en el diseño de un nuevo pattern para sumar este flujo a la biblioteca. La idea es colaborar con el designer que hizo el diseño original para crear todos los estados necesarios y casos de uso para documentar el componente y llevarlo a desarrollo.

La fecha que hablamos como deadline para tener este componente sería mitad de diciembre aproximadamente.

Cualquier otra cosa nos avisan!
